### PR TITLE
Migrate LLVMCPU vectorization to its own pass.

### PR DESCRIPTION
**Pull request by hanhanW**
The below related code are deprecated because they are no longer used by anyone.

- Deprecate SingleTilingExpert
- Deprecate StrategyVectorize pass

The VectorizationPattern and LinalgVectorizationPattern are not deleted because they are used by LLVMGPU.

### CODE REVIEW

File: compiler/src/iree/compiler/Codegen/LLVMCPU/LLVMCPUVectorization.cpp
Comment (hanhanW) [inline compiler/src/iree/compiler/Codegen/LLVMCPU/LLVMCPUVectorization.cpp:27]: The below static methods are copied from LinalgTensorCodegenDriver w/o any modification. I plan to add better utils to Codegen/Utils.h and LLVMCPU/Utils.h for handling the case.

**Review by MaheshRavishankar — COMMENTED** [review]
Looks good to me. Diego maybe has more context on this, so ill defer to him.

**Comment by dcaballe** [comment]
As we discussed, it would be great if I could land all the in-flight masking patches before we land this. Is it possible to wait?

**Comment by hanhanW** [comment]
> As we discussed, it would be great if I could land all the in-flight masking patches before we land this. Is it possible to wait?

Is there an ETA about it?

Sorry that I did not give you a heads-up. I was trying to see what the potential efforts would be. It's clearer to me after I prototype the PR.

I looked into other changes  that you sent out and found that we only have to move slight code from CodegenDriver file to LLVMCPUVectorization file. E.g., [compiler/src/iree/compiler/Codegen/Sandbox/LinalgTensorCodegenDriver.cpp](https://github.com/openxla/iree/pull/12604/files#diff-e105ec843efed97f5884ad26c85561c72967f0546029bd772335634f23553c31)

It adds a new `inferVectorSizesFromIR` method and it is called in `getVectorSize()`. This kind of moving is easy. Do I miss other cases? I can rebase on your PR and fix the conflicts if you can send me draft PRs and ETA.

**Comment by dcaballe** [comment]
Mostly blocked by https://reviews.llvm.org/D145681. @jpienaar is looking into it and I think it should land soon

**Review by dcaballe — APPROVED** [review]
Thanks!

**Pull request merged**
